File: app/src/main/java/com/sinthoras/randograf/cards/CardJoker.java
package com.sinthoras.randograf.cards;

import android.content.res.Resources;

import com.sinthoras.randograf.Colors;
import com.sinthoras.randograf.R;
import com.sinthoras.randograf.structure.Structure;
import com.sinthoras.randograf.structure.StructureGenerator;

public class CardJoker extends Card {

    private final Structure structure = StructureGenerator.generateStructure(1).withColor(Colors.JOKER);

    @Override
    public int getDuration() {
        return 0;
    }

    @Override
    public void paint() {
        // TODO
    }

    @Override
    public String getTitle() {
        return Resources.getSystem().getString(R.string.card_title_joker);
    }
}

File: app/src/main/java/com/sinthoras/randograf/cards/CardJoker.kt
package com.sinthoras.randograf.cards

import com.sinthoras.randograf.BlockColors
import com.sinthoras.randograf.structure.StructureGenerator
import com.sinthoras.randograf.R

class CardJoker : Card() {
    private val structure = StructureGenerator.generateStructure(1).withColor(BlockColors.ALL)

    fun getStructure() = structure

    override fun getDuration(): Int {
        return 0
    }

    override fun getTitle(): Int {
        return R.string.card_title_joker
    }
}
